### ISSUE TITLE
use Kronecker.jl for `kron`?

### ISSUE THREAD

**Issue by nickrobinson251** [JuliaStats/Distributions.jl#939]
https://github.com/MichielStock/Kronecker.jl

**Comment by johnczito** [comment]
I have a PR in mind for `PDMats.jl` that would create `kron(A::PDMat, B::PDMat) -> PDMat` so that we can get rid of things like this:

https://github.com/JuliaStats/Distributions.jl/blob/d00c780d680993519cf3dd0ce2b71dadd477afcf/src/matrix/matrixnormal.jl#L125

What other use cases do you have in mind?

**Comment by nickrobinson251** [comment]
I think all uses of `kron` could use this package to avoid actually computing the big "kron matrix" except when completely necessary. I think `kron` is only used in MatrixNormal and MatrixTDist. If we got rid of the `kron` call in MatrixNormal, then I guess it's only a question if this would be better for the MatrxiTDist use

**Comment by johnczito** [comment]
https://github.com/JuliaStats/Distributions.jl/blob/d00c780d680993519cf3dd0ce2b71dadd477afcf/src/matrix/matrixnormal.jl#L125

https://github.com/JuliaStats/Distributions.jl/blob/d00c780d680993519cf3dd0ce2b71dadd477afcf/src/matrix/matrixtdist.jl#L165

Completely agree. We are currently ignoring the fact that 
```
cholesky(A ⊗ B) = cholesky(A) ⊗ cholesky(B)
```
Since in both cases the two factors (`V`, `U` and `Σ`, `Ω`) are already `PDMat`, you should be able to go straight to the kronecker product as a `PDMat` instead of naively computing the full product and then computing its cholesky factor. But I think we will still need the full, dense kronecker product of the cholesky factors.

**Comment by nickrobinson251** [comment]
yeah, `cholesky` would be a good addition to Kronecker.jl (fyi https://github.com/MichielStock/Kronecker.jl/issues/10)

I think needing a `PDMat` is a reason to not use Kronecker.jl, but a better reason to just remove PDMats (I'm not sure they're worth the bother)

**Comment by johnczito** [comment]
Yeah, I will confess I don't understand what `PDMat` accomplishes that `Cholesky` doesn't (other than using twice the memory...).

**Comment by MichielStock** [comment]
Will try to add `cholesky` and others to `Kronecker.jl` today!

**Comment by johnczito** [comment]
@nickrobinson251 Since it's `PDMat`'s world and we're just living in it, we now have JuliaStats/PDMats.jl#100, which we can make use for the time being.

**Comment by nickrobinson251** [comment]
haha. Really nice work, @johnczito! Thanks :) 

Let's leave this open issue open, because the world may change one day?

**Comment by matbesancon** [comment]
ping @andreasnoack on this. I would prefer replacing PDMats with Cholesky for now. Kronecker looks promising but still recent stuff

**Comment by andreasnoack** [comment]
Yes. I'd be fine with not using PDMats. Does the `MatrixNormal` support any of the user PD types from PDMats?

**Comment by johnczito** [comment]
@andreasnoack Good point. Everything will work except the sampler. The sampler assumes that you have `PDMat` instances with a `chol` field and whatnot:

https://github.com/JuliaStats/Distributions.jl/blob/a15c8ba2ced774910d0bd6e47087946a7ac8824a/src/matrix/matrixnormal.jl#L118

I'll open an issue about this and work on generalizing it.

**Comment by johnczito** [comment]
Actually, JuliaStats/PDMats.jl#81 should fix this.
